packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2822,26 +2822,56 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         // The call to fetch the snapshot is very expensive and not always needed.
         // It should only be done by the summarizerNode, if required.
         const snapshotTreeFetcher = async () => {
+            // When fetching from storage we will always get the latest version and do not use the ackHandle.
             const fetchResult = await this.fetchSnapshotFromStorage(
-                ackHandle,
-                summaryLogger,
+             null,
+             summaryLogger,
+             {
+                 eventName: "RefreshLatestSummaryGetSnapshot",
+                 ackHandle,
+                 summaryRefSeq,
+                 fetchLatest: true,
+             });
+             return fetchResult.snapshotTree;
+         };
+         const result = await this.summarizerNode.refreshLatestSummary(
+             proposalHandle,
+             summaryRefSeq,
+             snapshotTreeFetcher,
+             readAndParseBlob,
+             summaryLogger,
+         );
+
+         // In case we had to retrieve the latest snapshot and it is different than summaryRefSeq,
+         // wait for the delta manager to catch up.
+         if (result.latestSummaryUpdated === true && result.wasSummaryTracked === false) {
+            assert(result.snapshot !== undefined, "should have tree result");
+            const latestSnapshotRefSeq = await seqFromTree(result.snapshot, readAndParseBlob);
+            if (latestSnapshotRefSeq !== summaryRefSeq) {
+                summaryLogger.sendTelemetryEvent(
                 {
-                    eventName: "RefreshLatestSummaryGetSnapshot",
-                    ackHandle,
-                    summaryRefSeq,
-                    fetchLatest: false,
+                    eventName: "LatestSummaryRetrieved",
+                    lastSequenceNumber: latestSnapshotRefSeq,
+                    targetSequenceNumber: summaryRefSeq,
                 });
-            return fetchResult.snapshotTree;
-        };
-        const result = await this.summarizerNode.refreshLatestSummary(
-            proposalHandle,
-            summaryRefSeq,
-            snapshotTreeFetcher,
-            readAndParseBlob,
-            summaryLogger,
-        );
+                if (latestSnapshotRefSeq > this.deltaManager.lastSequenceNumber) {
+                    // We need to catch up to the latest summary's reference sequence number before pausing.
+                    await PerformanceEvent.timedExecAsync(
+                        summaryLogger,
+                        {
+                            eventName: "WaitingForSeq",
+                            lastSequenceNumber: this.deltaManager.lastSequenceNumber,
+                            targetSequenceNumber: latestSnapshotRefSeq,
+                            lastKnownSeqNumber: this.deltaManager.lastKnownSeqNumber,
+                        },
+                        async () => waitForSeq(this.deltaManager, latestSnapshotRefSeq),
```

Comment:
It looks Ok to me, but please note that this code runs under lockedRefreshSummaryAckAction() lock. So summaries can't move forward. It looks fine, as summarizing in current state would fail due to us not using latest parent.